tests/expose_test.go

```diff
@@ -460,13 +437,73 @@ var _ = Describe("Expose", func() {
 				Expect(err).ToNot(HaveOccurred())
 
 				By("Waiting for the pod to report a successful connection attempt.")
-				getStatus = func() k8sv1.PodPhase {
-					pod, err := virtClient.CoreV1().Pods(job.Namespace).Get(job.Name, k8smetav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					return pod.Status.Phase
+				waitForJobCompletion(&virtClient, job, "success")
+			})
+
+			It("[test_id:343]Should Verify an exposed service of a VM is not functional after VM deletion.", func() {
+				By("Getting back the cluster IP given for the service")
+				svc, err := virtClient.CoreV1().Services(vm.Namespace).Get(serviceName, k8smetav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				serviceIP := svc.Spec.ClusterIP
+
+				By("Starting a pod which tries to reach the VMI via ClusterIP")
+				job := tests.NewHelloWorldJob(serviceIP, servicePort)
+				job, err = virtClient.CoreV1().Pods(vm.Namespace).Create(job)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Waiting for the pod to report a successful connection attempt")
+				waitForJobCompletion(&virtClient, job, "success")
+
+				By("Comparing the service's endpoints IP address to the VM pod IP address.")
+				// Get the IP address of the VM pod.
+				vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(vm.Name, &k8smetav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				vmPod := tests.GetRunningPodByVirtualMachineInstance(vmi, vmi.Namespace)
+				Expect(err).ToNot(HaveOccurred())
+				vmPodIpAddress := vmPod.Status.PodIP
+
+				// Get the IP address of the service's endpoint.
+				endpointsName := serviceName
+				svcEndpoints, err := virtClient.CoreV1().Endpoints(vm.Namespace).Get(endpointsName, k8smetav1.GetOptions{})
+				Expect(svcEndpoints).ToNot(BeNil())
+				endptSubsetIpAddress := func() string {
+					for _, epSubset := range svcEndpoints.Subsets {
+						for _, subsetAddr := range epSubset.Addresses {
+							return subsetAddr.IP
+						}
+					}
+					return ""
+				}
+				Eventually(endptSubsetIpAddress).Should(BeEquivalentTo(vmPodIpAddress))
+
+				By("Deleting the VM.")
+				Expect(virtClient.VirtualMachine(vm.Namespace).Delete(vm.Name, &k8smetav1.DeleteOptions{})).To(Succeed())
+				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
+
+				By("Verifying the service's endpoints IP address is deleted once the VM was deleted.")
+				svcEndpoints, err = virtClient.CoreV1().Endpoints(vm.Namespace).Get(endpointsName, k8smetav1.GetOptions{})
+				Expect(svcEndpoints).ToNot(BeNil())
+				endptSubsetIpAddress = func() string {
+					for _, epSubset := range svcEndpoints.Subsets {
+						for _, subsetAddr := range epSubset.Addresses {
+							return subsetAddr.IP
+						}
+					}
+					return ""
 				}
+				Eventually(endptSubsetIpAddress).ShouldNot(BeEquivalentTo(vmPodIpAddress))
+
+				svcEndpoints, err = virtClient.CoreV1().Endpoints(vm.Namespace).Get(endpointsName, k8smetav1.GetOptions{})
+				Expect(svcEndpoints).ToNot(BeNil())
```

Comment:
same comment as below

Reply:
My bad, I mistakenly added this check. It is incorrect and unnecessary.

Reply:
Done.